src/weaponstats.cc

```diff
@@ -0,0 +1,171 @@
+        }
+
+    short quality = random (100);
+
+    short ammoClip;
+    short damage;
+    float accuracy;
+    float moveSpeed;
+    float range;
+    int   properties;
+
+    if (weaponid == 18 || weaponid == 16 || weaponid == 39 || weaponid == 17)
+        {
+            properties = 0x100;
+        }
+    else if (weaponid == 35 || weaponid == 36)
+        {
+            properties = 0x48214;
+        }
+    else if (weaponid == 38)
+        {
+            properties = 0x238;
+        }
+    else
+        {
+            properties = hexFlags[random (0, 16)];
+        }
+
+    if (quality < 10)
+        {
+            ammoClip  = random (1, 15);
+            damage    = random (1, 25);
+            accuracy  = randomFloat (0.1, 0.5);
+            moveSpeed = randomFloat (0, 1.5);
+            range     = randomFloat (10, 30);
+        }
+    else if (quality < 85)
+        {
+            ammoClip  = random (1, 30);
+            damage    = random (1, 50);
+            accuracy  = randomFloat (0.1, 0.75);
+            moveSpeed = randomFloat (0.5, 2);
+            range     = randomFloat (20, 50);
+        }
+    else if (quality < 95)
+        {
+            ammoClip  = random (1, 50);
+            damage    = random (1, 75);
+            accuracy  = randomFloat (0.25, 1);
+            moveSpeed = randomFloat (1, 2.5);
+            range     = randomFloat (20, 75);
+        }
+    else
+        {
+            ammoClip  = random (1, 100);
+            damage    = random (1, 150);
+            accuracy  = randomFloat (1, 2);
+            moveSpeed = randomFloat (1.5, 3);
+            range     = randomFloat (20, 150);
+        }
+
+    float values[] = {randomFloat (0, 0.70f), randomFloat (0, 0.70f),
+                      randomFloat (0, 1.2f)};
+
+    std::sort (values, values + 3, std::greater<float> ());
+
+    for (int id : weaponids)
+        {
+            if (id == 0)
+                break;
+
+            int itemAddress = statsStart + itemSize * id;
+
+            injector::WriteMemory<int> (itemAddress + 0x18, properties);
+
+            injector::WriteMemory<short> (itemAddress + 0x20, ammoClip);
+
+            injector::WriteMemory<short> (itemAddress + 0x22, damage);
+
+            injector::WriteMemory<float> (itemAddress + 0x38, accuracy);
+
+            injector::WriteMemory<float> (itemAddress + 0x3C, moveSpeed);
+
+            injector::WriteMemory<float> (itemAddress + 0x04, range);
+
+            injector::WriteMemory<float> (itemAddress + 0x08, range);
+
+            injector::WriteMemory<float> (itemAddress + 0x40, values[2]);
+
+            injector::WriteMemory<float> (itemAddress + 0x44, values[0]);
+
+            injector::WriteMemory<float> (itemAddress + 0x48, values[1]);
+
+            injector::WriteMemory<float> (itemAddress + 0x4C, values[2]);
+
+            injector::WriteMemory<float> (itemAddress + 0x50, values[0]);
+
+            injector::WriteMemory<float> (itemAddress + 0x54, values[1]);
```

Comment:
So instead of the injector::WriteMemory, you would do
```cpp
aWeaponInfo[id]->damage = damage;
```
for example.

Here's a definition of CWeaponInfo you could use

```cpp
class CWeaponInfo
{
public:
    int          m_eWeaponFire;
    float        m_fRange;
    unsigned int m_nFiringRate;
    unsigned int m_nReload;
    unsigned int m_nAmountofAmmunition;
    unsigned int m_nDamage;
    float        m_fSpeed;
    float        m_fRadius;
    float        m_fLifespan;
    float        m_fSpread;
    CVector      m_vecFireOffset;
    unsigned int m_AnimToPlay;
    float        m_fAnimLoopStart;
    float        m_fAnimLoopEnd;
    float        m_fAnimFrameFire;
    float        m_fAnim2LoopStart;
    float        m_fAnim2LoopEnd;
    float        m_fAnim2FrameFire;
    float        m_fAnimBreakout;
    int          m_nModelId;
    int          m_nModel2Id;
    unsigned int m_nWeaponFlags;
    unsigned int m_WeaponSlot;
};
```